tensordict/prototype/tensorclass.py

```diff
@@ -187,10 +184,10 @@ def __torch_function__(
     cls.to_tensordict = _to_tensordict
     cls.device = property(_device, _device_setter)
     cls.batch_size = property(_batch_size, _batch_size_setter)
+    cls.__DATACLASS = True
```

Comment:
yeah, sure. typo.
is_dataclass uses the a similar trick checking the existence of _FIELDS. we could check for both ._tensordict and ._non_tensordict instead of the attribute. wdyt?

Reply:
Yes, I like the idea of checking for existing fields rather than adding a new field for the sole purpose of this check

Reply:
done in #301